packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -1135,24 +1214,77 @@ Widget _buildCupertinoDialogTransitions(BuildContext context, Animation<double>
 Future<T?> showCupertinoDialog<T>({
   required BuildContext context,
   required WidgetBuilder builder,
+  String? barrierLabel,
   bool useRootNavigator = true,
   bool barrierDismissible = false,
   RouteSettings? routeSettings,
 }) {
   assert(builder != null);
   assert(useRootNavigator != null);
-  return showGeneralDialog(
+
+  return Navigator.of(context, rootNavigator: useRootNavigator).push<T>(CupertinoDialogRoute<T>(
+    builder: builder,
     context: context,
     barrierDismissible: barrierDismissible,
-    barrierLabel: CupertinoLocalizations.of(context).modalBarrierDismissLabel,
+    barrierLabel: barrierLabel,
     barrierColor: CupertinoDynamicColor.resolve(_kModalBarrierColor, context),
+    settings: routeSettings,
+  ));
+}
+
+/// A dialog route that shows an iOS-style dialog.
+///
+/// Normally, [showCupertinoDialog] is used to display a Cupertino dialog.
```

Comment:
I would say "normally" here. I would just say, it's used internally by showCupertinoDialog or it can be pushed onto the Navigator stack directly to enable state restoration for the dialog, see [showCupertinoDialog] for an example. Or something like that.